### ISSUE TITLE
Object.assign

### ISSUE THREAD

**Issue by nelsonic** [dwyl/goodparts#221]
[`Object.assign`](
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) is an ***ES6*** feature I _really_ like because it is _elegant_.

```js
// we have 3 objects with various properties:
var user = { name: 'Jack', email: 'jackiscuel@email.net' };
var meta = { height: '185', eyes: 'blue', hair: 'blonde' };
var social = { twitter: 'jackyboy', instagram: 'iamjack' };
// and combine them into a _new_ object with Object.assign:
var jack = Object.assign({}, user, meta, social);
```
This feels like the "_right_" way of combining Objects ... (_why wasn't it **always** this way!?!?_)

Sadly, `Object.assign` is ***not available*** (_and will never be available_) in ***Internet Explorer***:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Browser_compatibility

The **ES5** version takes approx _30 seconds more_ to _type_ and requires an extra line of code:
```js
var jack = {};
[user, meta, social].forEach(function(o) { Object.keys(o).forEach(function(k) { jack[k] = o[k]; }); });
```
> Try it: https://repl.it/DfSg/3 (_can you simplify it?_)

Is it time to _investigate_ using Closure Compiler https://github.com/dwyl/learn-closure-compiler/issues/1 so we can use ES6? or can people _live_ with writing a few extra lines so we don't _have_ to "_transpile_"...?

**Comment by jrans** [comment]
hehe @nelsonic. Yes will have to look into asap

**Comment by des-des** [comment]
Without mutatation (dont do this)
```js
const pair2Str = ([k, v]) => "\"" + k + "\":" + JSON.stringify(v);
const entrySeq = o => Object.keys(o).map(k => ([k, o[k]]));
const entrySeq2JSON = entrySeq => '{' + entrySeq.map(pair2Str).join(',') + '}'
const merge = objArr => JSON.parse(entrySeq2JSON(objArr
  .reduce((wholeEntrySeq, obj) => wholeEntrySeq.concat(entrySeq(obj)), [])))
console.log(merge([{a: 1}, {b: 2}]))
```

**Comment by jackmoore** [comment]
I can't imagine not transpiling at this point, the ES6 features are such a big quality of life improvement.  I fell in love so fast with Object.assign, then dumped it for the spread operator (using Babel, I'm not sure if Closure Compiler supports that).

**Comment by nelsonic** [comment]
@jackmoore yeah, agree `Object.assign` feels _logical_ and is (_way_) more _beginner friendly_ than the ES5 alternative... As for the spread operator from my (_albeit limited_) experience I've seen beginners struggle with them `...`
